fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java

```diff
@@ -0,0 +1,229 @@
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.common.Config;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.MetaNotFoundException;
+import org.apache.doris.common.UserException;
+import org.apache.doris.common.util.LogBuilder;
+import org.apache.doris.common.util.LogKey;
+import org.apache.doris.load.BrokerFileGroup;
+import org.apache.doris.load.FailMsg;
+import org.apache.doris.load.PullLoadSourceInfo;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+
+public class BrokerLoadJob extends LoadJob {
+
+    private static final Logger LOG = LogManager.getLogger(BrokerLoadJob.class);
+
+    private BrokerDesc brokerDesc;
+    // include broker desc and data desc
+    private PullLoadSourceInfo dataSourceInfo = new PullLoadSourceInfo();
+
+    public BrokerLoadJob(long dbId, String label, BrokerDesc brokerDesc) {
+        super(dbId, label);
+        this.timeoutSecond = Config.pull_load_task_default_timeout_second;
+        this.brokerDesc = brokerDesc;
+    }
+
+    public static BrokerLoadJob fromLoadStmt(LoadStmt stmt) throws DdlException {
+        // get db id
+        String dbName = stmt.getLabel().getDbName();
+        Database db = Catalog.getInstance().getDb(stmt.getLabel().getDbName());
+        if (db == null) {
+            throw new DdlException("Database[" + dbName + "] does not exist");
+        }
+
+        // create job
+        BrokerLoadJob brokerLoadJob = new BrokerLoadJob(db.getId(), stmt.getLabel().getLabelName(),
+                                                        stmt.getBrokerDesc());
+        brokerLoadJob.setJobProperties(stmt.getProperties());
+        brokerLoadJob.checkDataSourceInfo(db, stmt.getDataDescriptions());
+        brokerLoadJob.setDataSourceInfo(db, stmt.getDataDescriptions());
+        return brokerLoadJob;
+    }
+
+    private void setDataSourceInfo(Database db, List<DataDescription> dataDescriptions) throws DdlException {
+        for (DataDescription dataDescription : dataDescriptions) {
+            BrokerFileGroup fileGroup = new BrokerFileGroup(dataDescription);
+            fileGroup.parse(db);
+            dataSourceInfo.addFileGroup(fileGroup);
+        }
+    }
+
+    @Override
+    public void executeScheduleJob() {
+        LoadTask task = new BrokerLoadPendingTask(this, dataSourceInfo.getIdToFileGroups(), brokerDesc);
+        tasks.add(task);
+        Catalog.getCurrentCatalog().getLoadManager().submitTask(task);
+    }
+
+    @Override
+    public void onTaskFinished(TaskAttachment attachment) {
+        if (attachment instanceof BrokerPendingTaskAttachment) {
+            onPendingTaskFinished((BrokerPendingTaskAttachment) attachment);
+        } else if (attachment instanceof BrokerLoadingTaskAttachment) {
+            onLoadingTaskFinished((BrokerLoadingTaskAttachment) attachment);
+        }
```

Comment:
Do you have some `state` or something to indicate whether or not this call is valid in your state machine?

For example this function can be called concurrently? or called multiple times for same task?

Reply:
Situation 1 (BrokerPendingTask ):There is only one pending task in BrokerLoad. So this function will not be called concurrently when pending task.
Situation 2 (LoadingTask): I add the writelock before onLoadingTaskFinished. So the loading task will be serial executed.